src/twisted/internet/test/test_unix.py

```diff
@@ -429,18 +443,94 @@ def _dataReceived(self, data):
 
         dataToSend = b'some data needs to be sent'
         fdsToSend = [sendSocket.fileno(), recvSocket.fileno()]
-        ancillary = [(SOL_SOCKET, SCM_RIGHTS, pack('ii', *fdsToSend))]
-        sendmsg(sendSocket, dataToSend, ancillary)
+        ancillary, expectedCount = ancillaryPacker(fdsToSend)
+        try:
+            sendmsg(sendSocket, dataToSend, ancillary)
+        except (socket.error, OSError) as e:
+            # Some platforms fail if ancillary contains more than
+            # one entry (Mac OS 10.9 and 10.10, for example).
+            # No point in continuing if sendmsg fails, skip the test.
+            raise SkipTest('sendmsg failed: %s' % (e,))
```

Comment:
Thanks for pointing that out. I somehow felt unconfortable raising the skip from within the test code. That helps understand my initial gut feeling... :)